addons/web/static/tests/fields/relational_fields_tests.js

```diff
@@ -9475,6 +9478,13 @@ QUnit.module('relational_fields', {
         assert.strictEqual(form.$('th').length, 2,
             "should be 2 columns in the one2many");
         form.$buttons.find('.o_form_button_edit').click();
+        form.$('.o_field_many2one[name="product_id"] input').click();
+        $('li.ui-menu-item a:contains(xpad)').trigger('mouseenter').click();
+        assert.strictEqual(form.$('th').length, 1,
+            "should be 1 column when the product_id is set");
+        $('.o_field_many2one[name="product_id"] input').val('').trigger('keyup');
```

Comment:
do not forget the form. before $: $(...) is a global selector, and form.$(...) is a selector restricted to the DOM of the form widget